pgmpy/base/DAG.py

```diff
@@ -1467,6 +1469,139 @@ def to_graphviz(self, plot_edge_strength=False):
 
         return agraph
 
+    def to_lavaan(self) -> str:
+        """
+        Convert the DAG to lavaan syntax representation.
+
+        The lavaan syntax represents structural equations where each line
+        shows a dependent variable regressed on its parents using the ~ operator.
+        Isolated nodes (nodes with no parents) are not included in the output.
+
+        Returns
+        -------
+        str
+            String representation of the DAG in lavaan syntax format.
+            Each line represents a regression equation where the dependent
+            variable is regressed on its parents.
+
+        Examples
+        --------
+        >>> from pgmpy.base import DAG
+        >>> dag = DAG([("X", "Y"), ("Z", "Y")])
+        >>> print(dag.to_lavaan())
+        Y ~ X + Z
+
+        >>> dag2 = DAG([("A", "B"), ("B", "C")])
+        >>> print(dag2.to_lavaan())
+        B ~ A
+        C ~ B
+
+        >>> # Empty DAG returns empty string
+        >>> empty_dag = DAG()
+        >>> print(empty_dag.to_lavaan())
+        ""
+
+        Notes
+        -----
+        - Node names are converted to string representations using str().
+        - If node names contain spaces or special characters, they will be used as-is.
+        - Users should ensure node names are valid in R/lavaan context if needed.
+
+        References
+        ----------
+        lavaan syntax: http://lavaan.ugent.be/tutorial/syntax1.html
+        """
+        lavaan_statements = []
+
+        # Iterate over each node and create equation for non-root nodes
+        for node in sorted(self.nodes(), key=str):
+            parents = self.get_parents(node)
+            if parents:  # If node has parents (i.e., not a root node)
+                # Convert node and parents to strings and sort parents
```

Comment:
I would suggest removing these comments as they are kind of obvious given the comment on the for loop.